include/moqt/core/base_track_handler.h

```diff
@@ -0,0 +1,138 @@
+/*
+ *  Copyright (C) 2024
+ *  Cisco Systems, Inc.
+ *  All Rights Reserved
+ */
+#pragma once
+
+#include <optional>
+#include <transport/span.h>
+#include <moqt/common.h>
+#include <vector>
+
+namespace moq::transport {
+
+    /**
+     * @brief MoQ track base handler for tracks (subscribe/publish)
+     *
+     * @details Base MoQ track handler
+     */
+    class BaseTrackHandler
+    {
+      public:
+        friend class Transport;
+
+        /**
+         * @brief Track mode object of object published or received
+         *
+         * @details QUIC stream handling mode used to send objects or how object was received
+         */
+        enum class TrackMode : uint8_t
+        {
+            kDatagram,
+            kStreamPerObject,
+            kStreamPerGroup,
+            kStreamPerTrack
+        };
+
+        virtual ~BaseTrackHandler() = default;
+
+        // --------------------------------------------------------------------------
+        // Public API methods that normally should not be overridden
+        // --------------------------------------------------------------------------
+
+        BaseTrackHandler() = delete;
+
+      protected:
+        /**
+         * @brief Track delegate constructor
+         *
+         * @param full_track_name       Full track name struct
+         */
+        BaseTrackHandler(const FullTrackName& full_track_name)
```

Comment:
i think we need a constructor to take in tuples for namespace , byte string for name